folly/FileUtil.cpp

```diff
@@ -35,7 +35,26 @@ namespace folly {
 using namespace fileutil_detail;
 
 int openNoInt(const char* name, int flags, mode_t mode) {
+#if defined(__ANDROID__)
+  // NDK bionic with FORTIFY has this definition:
+  // https://android.googlesource.com/platform/bionic/+/9349b9e51b/libc/include/bits/fortify/fcntl.h
+  // ```
+  // __BIONIC_ERROR_FUNCTION_VISIBILITY
+  // int open(const char* pathname, int flags, mode_t modes, ...) __overloadable
+  //         __errorattr(__open_too_many_args_error);
+  // ```
+  // This is originally to prevent open() with incorrect parameters.
+  //
+  // However, combined with folly wrapNotInt, template deduction will fail.
+  // In this case, we create a custom Open lambda to bypass the error.
+  // The solution is referenced from
+  // https://github.com/llvm/llvm-project/commit/0a0e411204a2baa520fd73a8d69b664f98b428ba
+  //
+  auto Open = [&]() { return open(name, flags, mode); };
```

Comment:
Let's write this as:

```
auto openWrap = +[](const char* pathname, int flags, mode_t modes) {
  return open(name, flags, mode);
}
auto openFunc = kIsAndroid ? openWrap : openFunc;
return int(wrapNoInt(openFunc, name, flags, mode));
```

Reply:
I think it would be simpler to just use the lambda in all cases, and don't bother checking `kIsAndroid`.  The use of the lambda doesn't appear to affect the code generated by either gcc or clang in optimized builds.

I think we can just say
```
auto openWrapper = [&] { return open(name, flags, mode); };
return int(wrapNoInt(openWrapper));
```

Reply:
thanks for the review.
should i revise changes in this pr, as imported to facebook internal phabricator already?

Reply:
No need to revise.

Reply:
Thanks!

Reply:
Is the import blocked due to some reason ? I see it has been blocked for two days.

Reply:
This is merged internally. The syncing from the internal repo to github is not working at the moment, but once it is working again this change will be merged on github as well.

Reply:
When this synced to github. Can you draft a new release for this ? react-native has been stuck for a while.

Reply:
@simpkins Looks the import is stuck.

Reply:
@simpkins ping